From 3bdbb3f5398685b30da8e80bd9d90b65e52ba3ca Mon Sep 17 00:00:00 2001
From: Voon Weifeng <weifeng.voon@intel.com>
Date: Tue, 21 Apr 2020 22:24:18 +0800
Subject: [PATCH 4/8] net: phy: fix Marvell 88E2110 PHY loopback

1.Added the missing disable PHY loopback function.
2.Updated to use phy_modify_mmd() to change the only required loopback bit.

Fixes: fcdd0b800b1e ("REVERTME: net: phy: Add phy loopback support for 88E2110 PHY")

Signed-off-by: Voon Weifeng <weifeng.voon@intel.com>
---
 drivers/net/phy/marvell10g.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/marvell10g.c b/drivers/net/phy/marvell10g.c
index 0d7ce35db5f4..159ed111ca22 100644
--- a/drivers/net/phy/marvell10g.c
+++ b/drivers/net/phy/marvell10g.c
@@ -554,8 +554,9 @@ static int m88e2110_soft_reset(struct phy_device *phydev)
 
 static int m88e2110_loopback(struct phy_device *phydev, bool enable)
 {
-	return phy_write_mmd(phydev, MDIO_MMD_PCS, MV_PCS_BASE_T,
-			     M88E2110_LOOPBACK);
+	return phy_modify_mmd(phydev, MDIO_MMD_PCS, MV_PCS_BASE_T,
+			  M88E2110_LOOPBACK,
+			  enable ? M88E2110_LOOPBACK : 0);
 }
 
 static struct phy_driver mv3310_drivers[] = {
-- 
2.17.1

